### PR TITLE
Gel layout Manager

### PR DESCRIPTION
**Pull request by FostUK**
Ready for review.

Next steps will be:
* Custom buttons / objects (e.g: so title balances with play button)
* Look at file name e.g: gel/layout ok?
* Layout.ts prob needs "de classing" so we follow a similar code layout in modules.
* Some overlap between gel.ts / layout.ts / scaler.ts  to DRY up.

### CODE REVIEW

File: src/core/gel/group.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import { Context } from "core/startup";
+import "phaser-ce";
+import {AccessibleButton} from "../accessible-button";
+
+const buttonPad = 24;
+
+class Group extends Phaser.Group {
```
Comment (martin-stone) [inline src/core/gel/group.ts]: Lots of `any`s in here, even where we have types defined (e.g. context and metrics).

File: src/core/gel/group.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            case "middle":
+                this.y = this.metrics.verticals[this.vPos] - this.height / 2;
+                break;
+            case "bottom":
+                this.y = this.metrics.verticals[this.vPos] - (this.height + this.metrics.pad);
+                break;
+        }
+    }
+
+    private getNextPosition() {
```
Comment (martin-stone) [inline src/core/gel/group.ts]: Would Phaser.Group.align() do any of this work for us? (Not read up on it in detail, so may not be helpful.)

File: src/core/gel/layout.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    //TODO destroy this object as well as group members.
+    public destroy() {
+        this.root.destroy();
+    }
+
+    private resize(width: number, height: number, scale: number, stageHeight: number) {
+        this.metrics = { width, height: stageHeight, scale };
+
+        // Border is padded to 2% of longest edge.
+        this.metrics.pad = Math.trunc(Math.max(this.metrics.width, this.metrics.height) * 0.02);
```
Comment (martin-stone) [inline src/core/gel/layout.ts]: Extract 0.02 to a constant?

File: src/components/results/results.ts [hunk 4]
```diff
@@ -18,11 +19,18 @@ class Results extends Screen {
...
+            this.game,
+            this.context,
+            ["home", "soundOn", "settings", "restart", "continue"],
+        );
+
+        this.layout.setAction("restart", this.startNewGame.bind(this));
     }
 
     private startNewGame() {
+        this.layout.destroy();
```
Comment (martin-stone) [inline src/components/results/results.ts]: If every `Screen` will need a `Layout` maybe the construction and destruction could be done in the `Screen` class.

File: src/core/gel/group.ts (hunk 1 above)
Comment (FostUK) [reply src/core/gel/group.ts]: Added the ones we have. Still need to make one for the button config.

File: src/components/results/results.ts (hunk 4 above)
Comment (FostUK) [reply src/components/results/results.ts]: Have moved the destroy into the screen with a check if the layout exists.

File: src/core/gel/layout.ts (hunk 3 above)
Comment (FostUK) [reply src/core/gel/layout.ts]: Fixed this.

**Pull request merged**